Plane: fix DO_CHANGE_SPEED airspeed to only impact AUTO and GUIDED modes

Hwurzburg opened the pull request:
-localizes DO_CHANGE_SPEED changes for airspeed target to only AUTO (need help on isolating GUIDED) instead of changing the param for cruise speed itself..
-change restricts DO_CHANGE_SPEED to only affect AUTO (non landing) speed and is reset to param value on Mode entry/rentry
-currently, it impacts FBWB/CRUISE/RTL after leaving AUTO and only reset to stored param on ARMING and after QPLANE land (which are not needed any longer since param is not fiddled and those been removed)
-posting as WIP to get feedback on approach...will SITL test before and after this week
-also, the groundspeed min and trim throttle need to be fixed also, I think, but first this

Hwurzburg commented:
Added a commit to change GUIDED mode to ignore speed changes while in AUTO

magicrub commented:
I've been talking with @samuelctabor about adding some better control for stuff like this: params are for aircraft flight-envelope and then multiple profiles for different flight scenarios. Then we would just call up "scenario 1" or whatever and that would be LAND or TAKEOFF or SOARING/INaThermal or "RTL as fast as you can" ect ect..

Hwurzburg commented:
@peterbarker why is this failing an unrelated  ROVER test for reboot? please restart

Hwurzburg commented:
fixed per devcal...renamed variable.....
now DO_CHANGE_SPEED impacts only AUTO and GUIDED modes....if off board guided is in use and it issues a speed change, that will be overrided if a DO_CHANGE_SPEED is issued until next guided mode entry....
issuing a DO_CHANGE_SPEED starts overriding TRIM_AIRSPEED_CM param changes while still in mode, those changes take effect again upon next AUTO or GUIDED mode rentry

tridge commented on ArduPlane/navigation.cpp:
```diff
@@ -182,22 +182,31 @@ void Plane::calc_airspeed_errors()
-        const float land_airspeed = SpdHgt_Controller->get_land_airspeed();
-        if (is_positive(land_airspeed)) {
-            target_airspeed_cm = land_airspeed * 100;
+    } else if (control_mode == &mode_guided && new_airspeed_cm >= 0) { //DO_CHANGE_SPEED overrides onboard guided speed commands, user would have re-enter guided mode to revert
```
lets make it > 0

Hwurzburg replied:
done

tridge commented on ArduPlane/navigation.cpp:
```diff
@@ -182,22 +182,31 @@ void Plane::calc_airspeed_errors()
-            // fallover to normal airspeed
-            target_airspeed_cm = aparm.airspeed_cruise_cm;
+            // normal AUTO mode and new_airspeed variable was set by DO_CHANGE_SPEED command while in AUTO mode
+            if (new_airspeed_cm !=-1) {
```
check for > 0

tridge replied:
can we change all the checks on new_airspeed_cm to be > 0 please

Hwurzburg replied:
fixed

tridge commented on ArduPlane/mode_auto.cpp:
```diff
@@ -11,6 +11,8 @@ bool ModeAuto::_enter()
     // start or resume the mission, based on MIS_AUTORESET
     plane.mission.start_or_resume();
+    // initialize speed variable used in Auto for DO_CHANGE_SPEED commands
+    plane.new_airspeed_cm = -1;
```
also set to -1 in the exit for AUTO and GUIDED

tridge replied:
add it to Mode::enter() to cover all modes

Hwurzburg replied:
done

Hwurzburg commented:
@peterbarker made the changes...but will need help in understanding why EKF lane switches would be involved in anything that is in this PR....

Hwurzburg commented:
fixed autotests....were expecting CRUISE at ARSPD_FBW_MAX (error introduced with off board guided)...but fixed here...had to change to account for CRUISE at trim_arspd_cm now

peterbarker commented:
@jaxxzer @patrickelectric did we modify some sub tests recently?

Williangalvani commented:
> did we modify some sub tests recently?

Not recently, the failing test was introduced back in August.
